mentoring/mentoring.py

```diff
@@ -83,54 +65,105 @@ class MentoringBlock(XBlockWithLightChildren, StepParentMixin):
-    followed_by = String(help="url_name of the step after the current mentoring block in workflow",
-                         default=None, scope=Scope.content)
-    url_name = String(help="Name of the current step, used for URL building",
-                      default='mentoring-default', scope=Scope.content)
-    enforce_dependency = Boolean(help="Should the next step be the current block to complete?",
-                                 default=False, scope=Scope.content, enforce_type=True)
+        return value == default_xml_content
+
+    # Content
+    MENTORING_MODES = ('standard', 'assessment')
+    mode = String(
+        help="Mode of the mentoring. 'standard' or 'assessment'",
+        default='standard',
+        scope=Scope.content,
+        values=MENTORING_MODES
+    )
+    followed_by = String(
+        help="url_name of the step after the current mentoring block in workflow.",
+        default=None,
+        scope=Scope.content
+    )
+    max_attempts = Integer(
+        help="Number of max attempts for this questions",
+        default=0,
+        scope=Scope.content,
+        enforce_type=True
+    )
+    url_name = String(
+        help="Name of the current step, used for URL building",
+        default='mentoring-default',  # TODO in future: set this to xblock.fields.UNIQUE_ID and remove self.url_name_with_default
+        scope=Scope.content
+    )
+    enforce_dependency = Boolean(
+        help="Should the next step be the current block to complete?",
+        default=False,
+        scope=Scope.content,
+        enforce_type=True
+    )
     display_submit = Boolean(help="Allow to submit current block?", default=True, scope=Scope.content)
-    xml_content = String(help="XML content", default=_default_xml_content, scope=Scope.content)
-    weight = Float(help="Defines the maximum total grade of the block.",
-                   default=1, scope=Scope.content, enforce_type=True)
-    num_attempts = Integer(help="Number of attempts a user has answered for this questions",
-                           default=0, scope=Scope.user_state, enforce_type=True)
-    max_attempts = Integer(help="Number of max attempts for this questions", default=0,
-                           scope=Scope.content, enforce_type=True)
-    mode = String(help="Mode of the mentoring. 'standard' or 'assessment'",
-                  default='standard', scope=Scope.content)
-    step = Integer(help="Keep track of the student assessment progress.",
-                   default=0, scope=Scope.user_state, enforce_type=True)
-    student_results = List(help="Store results of student choices.", default=[],
-                           scope=Scope.user_state)
-
-    display_name = String(help="Display name of the component", default="Mentoring XBlock",
-                          scope=Scope.settings)
+    xml_content = String(help="XML content", default=default_xml_content, scope=Scope.content)
```

Comment:
@bradenmacdonald @antoviaque I've been seeing this thorn in our side keep coming up. Do we even need to bother with `url_name`? That's automatically generated at random by the CMS. Why do we handle it manually here? If it's omitted from the XML, it will still be set under the hood. Is there something special we're doing with it?